### PR TITLE
+str #23032 MergePreferred as Source, proposal

### PR DESCRIPTION
**Pull request by ktoso**
Refs https://github.com/akka/akka/issues/23032

We would have to do this for all merges, pretty simple to do and much nicer to use.

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/MergePreferredSpec.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+import akka.stream.testkit.TwoStreamsSetup
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class MergePreferredSpec extends TwoStreamsSetup {
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/MergePreferredSpec.scala]: Made a separate file, but would live together with the GraphMergePreferredSpec I guess

**Comment by 2m** [comment]
We have a more general `Source.combine` which works with any `UniformFanInShape`. The test case here can be rewritten like

```scala
val result = Source.combine(preferred, aux)(allPorts ⇒ new MergePreferred(allPorts - 1, eagerComplete = false))
  .grouped(numElements * 2)
  .runWith(Sink.head[Seq[Int]])
```

**Comment by 2m** [comment]
I think `Source.combine` achieves the goal of using any fan in stage without resorting to the Graph DSL. It also general and works for user defined stages. Therefore I would be against hardcoding this to the build-in fan in stages.

**Comment by ktoso** [comment]
If we decide to not add these convenience things, then we should add a section on actually using that API in such ways, I don't think people are aware of this (esp in more advanced usage patterns).

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/MergePreferredSpec.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  "preferred merge" must {
+
+    "prefer selected input more than others" in {
+      val numElements = 10000
+
+      val preferred = Source(Stream.fill(numElements)(1))
+      val aux = Source(Stream.fill(numElements)(2))
+
+      val result = MergePreferred(preferred, aux)
```
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/MergePreferredSpec.scala]: This is an entirely different style from all the other factories though, shouldn't it rather be `Source.mergePreferred(pref, aux)` or possibly `prefSource.mergePreferred(aux)`?

Comment (patriknw) [reply akka-stream-tests/src/test/scala/akka/stream/scaladsl/MergePreferredSpec.scala]: I agree, probably also related to previous discussion about `combine`

**Comment by ktoso** [comment]
Closing this

**Comment by ktoso** [comment]
Closing this, can't even remember what it was...

**Pull request closed without merging**